core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,106 @@
+ */
+
+case class KubernetesCluster(id: KubernetesClusterLeoId,
+                             googleProject: GoogleProject,
+                             clusterName: KubernetesClusterName,
+                             location: Location,
+                             status: KubernetesClusterStatus,
+                             serviceAccountInfo: WorkbenchEmail,
+                             samResourceId: KubernetesClusterSamResource,
+                             auditInfo: AuditInfo,
+                             asyncFields: KubernetesClusterAsyncFields,
+                             namespaces: Set[KubernetesNamespaceName],
+                             labels: LabelMap,
+                             nodepools: Set[Nodepool]
+                            )
+
+object KubernetesCluster {
+  implicit class EnrichedKubernetesCluster(cluster: KubernetesCluster) {
+    def getGKEClusterId: KubernetesClusterId = KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
+  }
+}
+
+case class SaveKubernetesCluster(googleProject: GoogleProject,
+                                 clusterName: KubernetesClusterName,
+                                 location: Location,
+                                 status: KubernetesClusterStatus,
+                                 serviceAccountInfo: WorkbenchEmail,
+                                 samResourceId: KubernetesClusterSamResource,
+                                 auditInfo: AuditInfo,
+                                 labels: LabelMap,
+                                 initialNodepool: Nodepool) //the clusterId specified here isn't used, and will be replaced by the id of cluster saved beforehand
+
+case class KubernetesClusterSamResource(resourceId: String)
+
+case class KubernetesClusterAsyncFields(apiServerIp: Option[KubernetesApiServerIp],
+                                        networkInfo: Option[NetworkFields]
+                                       )
+
+case class NetworkFields(networkName: NetworkName,
+                         subNetworkName: SubnetworkName,
+                         subNetworkIpRange: IpRange)
+
+
+//TODO: possibly consolidate Kubernetes status when monitoring is implemented
+sealed trait KubernetesClusterStatus extends EnumEntry with Product with Serializable
+object KubernetesClusterStatus extends Enum[KubernetesClusterStatus] {
+  val values = findValues
+
+  // NOTE: Remember to update the definition of this enum in Swagger when you add new ones
+  case object Status_Unspecified extends KubernetesClusterStatus
+  case object Provisioning extends KubernetesClusterStatus
+  case object Running extends KubernetesClusterStatus
+  case object Reconciling extends KubernetesClusterStatus
+  case object Stopping extends KubernetesClusterStatus
+  case object Error extends KubernetesClusterStatus
+  case object Degraded extends KubernetesClusterStatus
+}
+
+//TODO: possibly consolidate Kubernetes status when monitoring is implemented
```

Comment:
```suggestion
//TODO: possibly consolidate Nodepool status when monitoring is implemented
```

Reply:
The intent here is that it would be nice to consolidate the statuses we have here such that we don't have to do too much work reimplementing monitoring, since both Kubernetes Nodepools and Clusters use google `Operation`s, which GCE also uses

Reply:
I'm still trying to understand this. Perhaps we can talk offline.